### DISCUSSION TITLE
Improve interaction with prettier and other markdown formatters

### DISCUSSION THREAD

**Discussion by Midnighter** [squidfunk/mkdocs-material#5528] [Q&A]
I like to use a formatter for markdown files, for example, prettier. Generally, this works really well since mostly we are only using markdown. However, prettier doesn't play well with references to classes. I can create references using `[MyClass][package.MyClass]` but prettier insists on making everything lowercase in the second pair of brackets. This is a little annoying since it means I can't use pre-commit hooks for those files where I use this.

Has anyone experienced the same and found a good fix? Thanks for any input.

**Comment by vladfaust** [comment]
I hate when people break standards and this becomes a norm.

**Reply by squidfunk** [reply]
@vladfaust Python Markdown is based on Gruber's official Markdown implementation, which wasn't standardized, but which [predates standardized implementations like CommonMark](https://github.com/Python-Markdown/markdown/blob/23c301de28e12426408656efdfa153b11d4ff558/docs/change_log/index.md) (in the case of CommonMark it's 10 years). We use Python Markdown because MkDocs uses it, as CommonMark was probably rather new at that time. However, I agree that third party tooling support could be better, which is why we're evaluating how to improve it as part of our efforts on [foundational work](https://squidfunk.github.io/mkdocs-material/blog/2024/08/19/how-were-transforming-material-for-mkdocs/) ☺️

**Reply by vladfaust** [reply]
Thank you for a thorough and educational response, @squidfunk. I'm sorry for my arrogance.
